modules/cli/src/main/scala/scala/cli/commands/Run.scala

```diff
@@ -103,11 +103,36 @@ object Run extends ScalaCommand[RunOptions] with BuildCommandHelpers {
   }
 
   def run(
-    options: RunOptions,
+    opts: RunOptions,
     inputArgs: Seq[String],
     programArgs: Seq[String],
     defaultInputs: () => Option[Inputs]
   ): Unit = {
+    val options =
+      if (!opts.sharedRun.sharedDebug.debug)
+        opts
+      else
+        val server = opts.sharedRun.sharedDebug.debugMode match {
+          case Some("attach") | Some("att") | Some("a") | None => "y"
+          case Some("listen") | Some("lis") | Some("l")        => "n"
+          case Some(a) =>
+            opts.shared.logger.message(
+              s"Warning: Unrecognized debug mode: $a. Switching to default attach mode."
+            )
+            "y"
+        }
+        val port = opts.sharedRun.sharedDebug.debugPort.getOrElse("5005")
+        opts.copy(
+          sharedRun = opts.sharedRun.copy(
+            sharedJava = opts.sharedRun.sharedJava.copy(
+              javaOpt = opts.sharedRun.sharedJava.javaOpt ++ List(
+                "-Xmx512m",
+                "-Xms128m",
```

Comment:
Scala CLI shouldn't apply this java options by default. I think, we can skip it.